ILTransform/TestProject.cs

```diff
@@ -223,15 +223,21 @@ public static bool MakePublic(bool isILTest, ref string line, bool force)
             if (force)
             {
                 int charIndex = line.SkipWhiteSpace();
+                string preffix = line[..charIndex];
                 if (isILTest)
                 {
+                    // They always start like .method or .class but might end without spaces
                     charIndex = line.SkipNonWhiteSpace(charIndex);
+                    preffix = string.Concat(line.AsSpan(0, charIndex), " "); // Add space and move index
                     if (charIndex < line.Length)
                     {
                         charIndex++;
                     }
                 }
-                line = string.Concat(line.AsSpan(0, charIndex), "public ", line.AsSpan(charIndex));
+                // Avoid white space at the end if line is ".class" or add white space if something continues
+                // after the new "public" token.
+                string suffix = charIndex == line.Length ? "" : string.Concat(" ", line.AsSpan(charIndex));
+                line = string.Concat(preffix, "public", suffix);
```

Comment:
This method seems messy trying to combine the C# and IL logic a bit even before your change..

if (IL)
  skip non whitespace
  insert " public"
else
  don't skip
  insert "public "

(btw, string.Insert(x, value) seems simpler than the Concat of prefix, value, suffix though that Concat pattern seems to exist in a lot of places)

(also, "prefix")

Reply:
I like your idea, thanks!